mongox/models.py

```diff
@@ -434,3 +434,18 @@ async def save(self: T) -> T:
         for k, v in self.dict(exclude={"id"}).items():
             setattr(self, k, v)
         return self
+
+    @classmethod
+    async def get_by_id(cls: typing.Type[T], id: typing.Union[str, bson.ObjectId]) -> T:
+        """
+        Get document by id
+        """
+
+        if isinstance(id, str):
+            try:
+                id = bson.ObjectId(id)
+            except bson.errors.InvalidId as e:
+                raise NoMatchFound() from e
```

Comment:
I'm not sure should raise NoMatchFound when id invalid, for the user, the final result is same: item does not exist

Reply:
Maybe worth creating another exception `InvalidObjectIdException(f'"{id}" is not a valid BSON ObjectId')`.
What do you think?